packages/frontend/src/pages/Users/components/UsersTableRow.tsx

```diff
@@ -0,0 +1,37 @@
+import { useHistory } from 'react-router-dom';
+import { UserDto, UserRole } from 'api/dist/user/types';
+import RoleBadge from './RoleBadge';
+import { UserAvatar } from '@/components/user/UserAvatar';
+import { shortenEthAddress } from '@/utils/index';
+import { getUsername } from '@/utils/users';
+
+interface IUsersTableRow {
+  data: UserDto;
+}
+
+const UsersTableRow = ({ data }: IUsersTableRow): JSX.Element => {
+  const history = useHistory();
+  return (
+    <div
+      className="my-3 px-4 flex justify-between items-center cursor-pointer h-12 rounded-md hover:bg-gray-100"
+      onClick={(): void => history.push(`pool/${data._id}`)}
+    >
+      <div className="w-1/3 flex items-center gap-4">
+        <UserAvatar user={data} />
+        <span className=" font-mono text-sm">
+          {shortenEthAddress(data.ethereumAddress!)}
```

Comment:
The user.ethereumAddress should always be defined (according to the mongoose entity). 

Thus I think we can should modify the User & UserDto types to reflect this (make ethereumAddresss no longer optional). Then you won't need this non-null assertion.

Reply:
Doesn't this collide with the needs of #361? If the User is created before account activation we don't know the eth address.

Reply:
> The user.ethereumAddress should always be defined (according to the mongoose entity).

That is outside the scope of this PR. When we're ready to address #361 there will need to be changes throughout the codebase.